pkg/lvm/lvm_util.go

```diff
@@ -265,3 +271,82 @@ func DestroySnapshot(snap *apis.LVMSnapshot) error {
 func getLVMSnapName(snapName string) string {
 	return strings.TrimPrefix(snapName, "snapshot-")
 }
+
+func decodeVgsJSON(raw []byte) ([]apis.VolumeGroup, error) {
+	output := &struct {
+		Report []struct {
+			VolumeGroups []map[string]string `json:"vg"`
+		} `json:"report"`
+	}{}
+	var err error
+	if err = json.Unmarshal(raw, output); err != nil {
+		return nil, err
+	}
+
+	if len(output.Report) != 1 {
+		return nil, fmt.Errorf("expected exactly one lvm report")
+	}
+
+	items := output.Report[0].VolumeGroups
+	vgs := make([]apis.VolumeGroup, 0, len(items))
+	for _, item := range items {
+		var vg apis.VolumeGroup
+		if vg, err = parseVolumeGroup(item); err != nil {
+			return vgs, err
+		}
+		vgs = append(vgs, vg)
+	}
+	return vgs, nil
+}
+
+func parseVolumeGroup(m map[string]string) (apis.VolumeGroup, error) {
+	var vg apis.VolumeGroup
+	vg.Name = m["vg_name"]
+	vg.UUID = m["vg_uuid"]
+
+	int32Map := map[string]*int32{
+		"pv_count": &vg.PVCount,
+		"lv_count": &vg.LVCount,
+	}
+	for key, value := range int32Map {
+		count, err := strconv.Atoi(m[key])
+		if err != nil {
+			err = fmt.Errorf("invalid format of %v=%v for vg %v: %v", key, m[key], vg.Name, err)
+		}
+		*value = int32(count)
+	}
+
+	resQuantityMap := map[string]*resource.Quantity{
+		"vg_size": &vg.Size,
+		"vg_free": &vg.Free,
+	}
+
+	for key, value := range resQuantityMap {
+		sizeBytes, err := strconv.ParseInt(
+			strings.TrimSuffix(strings.ToLower(m[key]), "b"),
+			10, 64)
+		if err != nil {
+			err = fmt.Errorf("invalid format of %v=%v for vg %v: %v", key, m[key], vg.Name, err)
+		}
+		quantity := resource.NewQuantity(sizeBytes, resource.BinarySI)
+		*value = *quantity //
+	}
+	return vg, nil
+}
+
+// ListLVMVolumeGroup invokes `vgs` to list all the available volume
+// groups in the node.
+func ListLVMVolumeGroup() ([]apis.VolumeGroup, error) {
+	args := []string{
+		"--options", "vg_all",
+		"--reportformat", "json",
```

Comment:
Need to add a minimum system requirement, because in older versions of Ubuntu 16.04 , `reportformat` option is not supported.

Reply:
Thanks for pointing out. In that case, do we need to support 16.04? It's getting EOLed in april 21. We can take a call of supporting only lvm versions greater than certain value. As per the [RHEL documentation](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/logical_volume_manager_administration/custom_report#json-output), it's being supported since redhat v7.3. I need to check how does it gets translated to lvm driver version. If reportformat flag is not supported, I presume that command will fail and user will not be able to use this capacity tracking feature. Parsing the default output format (tabular) will result in more tedious exercise & i'm not sure if it's worth the effort since 16.04 will be eoled in two months.

Reply:
@pawanpraka1 Can we decide on the minimum supported version

Reply:
This wont be an issue as we are using the LVM binary from the alpine image. So the version will not be a problem.